code/modules/multiz/hoist.dm

```diff
@@ -0,0 +1,210 @@
+///////////////////////////
+// Dost thou even hoist? //
+///////////////////////////
+
+/obj/item/hoist_kit
+	name = "hoist kit"
+	desc = "A setup kit for a hoist that can be used to lift things. The hoist will deploy in the direction you're facing."
+	icon = 'icons/obj/hoists.dmi'
+	icon_state = "hoist_case"
+
+/obj/item/hoist_kit/attack_self(mob/user)
+	new /obj/structure/hoist (get_turf(user), user.dir)
+	user.visible_message(span("warning", "[user] deploys the hoist kit!"), span("notice", "You deploy the hoist kit!"), span("notice", "You hear the sound of parts snapping into place."))
+	qdel(src)
+
+/obj/effect/hoist_hook
+	name = "hoist clamp"
+	desc = "A clamp used to lift people or things."
+	icon = 'icons/obj/hoists.dmi'
+	icon_state = "hoist_hook"
+	var/obj/structure/hoist/source_hoist
+	anchored = 1
+
+/obj/effect/hoist_hook/MouseDrop_T(atom/movable/AM,mob/user)
+	if(!istype(AM))
+		log_debug("AAAAAAA")
+		return
+	if (!AM.simulated || AM.anchored || source_hoist.hoistee || !Adjacent(AM))
+		return
+	source_hoist.attach_hoistee(AM)
+	user.visible_message(span("danger", "[user] attaches \the [AM] to the hoist clamp."), span("danger", "You attach \the [AM] to the hoist clamp."), span("danger", "You hear something clamp into place."))
+
+/obj/structure/hoist/proc/attach_hoistee(atom/movable/AM)
+	if (get_turf(AM) != get_turf(src))
+		AM.forceMove(get_turf(src))
+	hoistee = AM
+	if(ismob(AM))
+		var/mob/M = AM
+		M.captured = 1
+		M.update_canmove()
+	else
+		AM.anchored = 1
+
+/obj/effect/hoist_hook/MouseDrop(atom/dest)
+	if(!usr || !dest) return
```

Comment:
Nope! That's taken directly from the default MouseDrop proc. It's necessary for other `MouseDrop_T` things to work. And I can't directly call it because it doesn't return any info.